Tribler/Core/Modules/restapi/channel_endpoint.py

```diff
@@ -0,0 +1,100 @@
+import json
+import time
+
+from twisted.web import http, resource
+
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST
+from Tribler.community.allchannel.community import AllChannelCommunity
+
+
+VOTE_UNSUBSCRIBE = 0
+VOTE_SUBSCRIBE = 2
+
+
+class ChannelEndpoint(resource.Resource):
+    """
+    This endpoint is the root endpoint responsible for handing all requests regarding a channel in Tribler.
+    """
+
+    def __init__(self, session):
+        resource.Resource.__init__(self)
+        self.session = session
+
+    def getChild(self, cid, request):
+        return SpecificChannelEndpoint(self.session, cid)
+
+
+class SpecificChannelEndpoint(resource.Resource):
+    """
+    This class delegates specific requests (i.e. fetching torrents in a channel or subscribing)
+    to child endpoint handlers.
+    """
+
+    def __init__(self, session, cid):
+        resource.Resource.__init__(self)
+        self.session = session
+        self.cid = cid
+
+        child_handler_dict = {"subscribe": ChannelSubscribeEndpoint}
+        for path, child_cls in child_handler_dict.iteritems():
+            self.putChild(path, child_cls(self.session, self.cid))
+
+
+class SpecificChannelBaseEndpoint(resource.Resource):
+    """
+    Base class used for requests in a specific channel. Contains some utility methods that are commonly used
+    by child endpoint handlers.
+    """
+
+    def __init__(self, session, cid):
+        resource.Resource.__init__(self)
+        self.session = session
+        self.channel_db_handler = session.open_dbhandler(NTFY_CHANNELCAST)
+        self.cid = bytes(cid.decode('hex'))
+
+    @staticmethod
+    def return_404(request):
+        """
+        Returns a 404 response code if your channel has not been created.
+        """
+        request.setResponseCode(http.NOT_FOUND)
+        return "the channel with the provided cid does not exist"
+
+    def get_channel_from_db(self):
+        """
+        Returns information about the channel from the database. Returns None if the channel with given cid
+        does not exist.
+        """
+        channels_list = self.channel_db_handler.getChannelsByCID([self.cid])
+        if len(channels_list) == 0:
```

Comment:
I believe this is the more pythonic way: `if not channels_list` (checks for empty if it's a list or if it's none it also triggers)